src/main/java/edu/umich/its/cpm/MigrationController.java

```diff
@@ -1623,91 +1661,169 @@ public Response getSiteMigrationInBulkUpload(
+			siteName, toolId, toolName,
+			googleGroupId, googleGroupName, userId);
+		
+		// 3. delegate the actual message migrations to async calls
+		// Concatenating all site owners' names together with 
+		HashMap<String, String> status = migrationTaskService.processAddEmailMessages(
+				request, response, Utils.MIGRATION_TYPE_GOOGLE, userId,
+				new HashMap<String, String>(), googleGroupId, siteId,
+				toolId, saveBulkMigration);
+		
+		if (status.containsKey("errorMessage")) {
+			log.info(this + " batch upload call for site id="
+					+ siteId + " error message="
+					+ status.get("errorMessage"));
+		} else if (status.containsKey("migrationId")) {
+			log.info(this + " batch upload call for site id="
+					+ siteId + " migration started id="
+					+ status.get("migrationId"));
+		}
+		
+	}
+	
+	/**
+	 * 
+	 * @param sessionId
+	 * @param request
+	 * @param response
+	 * @param userId
+	 * @param bulkMigrationName
+	 * @param bulkMigrationId
+	 * @param siteId
+	 * @param siteName
+	 * @param toolId
+	 * @param toolName
+	 */
+	private void handleBulkResourceBoxMigration(String sessionId,
+			HttpServletRequest request, HttpServletResponse response,
+			String userId, String bulkMigrationName, String bulkMigrationId,
+			String siteId, String siteName, String toolId, String toolName) {
+
+		// get box folder id
+		String remoteUserId = Utils.getRemoteUser(request);
+		String boxAdminClientId = BoxUtils.getBoxClientId(remoteUserId);
+		String boxAdminClientSecret = BoxUtils.getBoxClientSecret(remoteUserId);
+		String boxSiteFolderName = "CTools - " + siteName;
+		
+		Info boxFolder = BoxUtils.createNewFolderAtRootLevel(userId,
+				boxAdminClientId, 
+				boxAdminClientSecret,
+				boxSiteFolderName, siteId, uRepository);
+		
+		// save the migration record into database
+		// the string to hold on all site owner's id, 
+		// and the admin id who are doing bulk migration is listed first
+		StringBuffer allSiteOwners = new StringBuffer(userId);
+		if (boxFolder != null) {
+			String boxFolderId = boxFolder.getID();
+			String boxFolderName = boxFolder.getName();
+
+			// add site members to Box folder as collaborators
+			HashMap<String, String> userRoles = get_site_members(siteId, sessionId);
+			for (String userEid : userRoles.keySet()) {
+				String userRole = userRoles.get(userEid);
+				String userEmail = Utils.getUserEmailFromUserId(userEid);
+				try
+				{
+					BoxUtils.addCollaboration(
+							env.getProperty(Utils.BOX_ADMIN_ACCOUNT_ID),
+							userEmail, userRole, boxFolderId,
+							boxAdminClientId, boxAdminClientSecret, uRepository);
+				}
+				catch (RestClientException e)
+				{
+					log.error("Problem retrieving site members for site id = " + siteId + " " + e.getStackTrace());;
+				}
+				catch (JSONException e)
+				{
+					log.error("Problem retrieving site members for site id = " + siteId + " " + e.getStackTrace());
```

Comment:
How about "Problem parsing ...." so we don't lose all the information in the exception?